packages/rpc-graphql/src/schema/ast/index.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { buildAccountGraphQLTypeDef, buildInstructionGraphQLTypeDef, buildTypeGraphQLTypeDef } from './type-defs';
+import { ProgramAst, ProgramAstSource } from './types';
+
+export interface AstConfig {
+    resolvers: Record<string, unknown>;
+    typeDefs: string;
+    resolveTypeExtension?: (obj: unknown) => string;
+}
+
+export interface SchemaAstConfig {
+    accounts: AstConfig;
+    instructions: AstConfig;
+    types: AstConfig;
+}
+
+function buildSchemaAstConfig(programAst: ProgramAstSource): SchemaAstConfig {
+    const parsedAst: ProgramAst = typeof programAst === 'string' ? JSON.parse(programAst) : programAst;
+    return {
+        accounts: {
+            resolvers: {},
+            typeDefs: parsedAst.accounts.map(buildAccountGraphQLTypeDef).join('\n'),
+            // TODO: Conditional function for mapping decoded data.
+            resolveTypeExtension: (_account): string => null as unknown as string,
```

Comment:
This would basically be that conditional mapping, taking a look at the parsed data and determining which type it is. Haven't 100% thought about how this would work, but ideally it should align with the base `jsonParsed` format before being transformed into it's proper GraphQL type.